SayTheirNames/Source/Controller/Home/Person/PersonController.swift

```diff
@@ -153,12 +160,24 @@ class PersonController: UIViewController {
         super.viewDidLoad()
         view.accessibilityIdentifier = "personView"
         self.network.fetchPersonDetails(with: person.identifier) { [weak self] result in
+            guard let self = self else { return }
+            
             switch result {
             case .success(let response):
-                self?.configure(with: response.person)
+                self.configure(with: response.person)
+                
+                guard self.person.donations.first != nil else {
+                    self.donationButtonContainerView.removeFromSuperview()
+                    var viewConstraint: NSLayoutConstraint?
+                    viewConstraint = self.tableView.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor)
+                    viewConstraint?.isActive = true
+                    return self.viewBlur.removeFromSuperview()
+                   }
+               
```

Comment:
Nice. I haven't tested it on various devices yet. I think it makes sense to move all this logic into its own function that can get called once we have all the data. Maybe even get called in the `configure` method. That keeps the results handling code super clean too. 

We might have some helpers already to update constraints (might not). Check the code in like `setupDonationBottomView ` and click into the `anchor` method, we might have something easier to use there.

I think the `self.viewBlur.removeFromSuperview()` should come one line above the `return` .. But also since you have it below, you probably don't even need it here at all since it gets called regardless of success or failure